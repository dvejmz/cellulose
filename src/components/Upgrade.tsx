import React from 'react';

export interface UpgradeProps {
  name: string;
  cost: number;
  currency: string;
};

const Upgrade: React.FC<UpgradeProps> = (props: UpgradeProps) => (
  <div className="upgrade" data-test-id="upgrade-ppc-2x">
    <button
      className="btn"
    >
      {props.name} ({props.currency}{props.cost})
    </button>
  </div>
);

export default Upgrade;
