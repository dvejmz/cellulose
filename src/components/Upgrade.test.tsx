import { shallow } from 'enzyme';
import React from 'react';
import Upgrade, { UpgradeProps } from './Upgrade';

describe('<Upgrade />', () => {
  let upgrade: any;

  beforeEach(() => {
    const props: UpgradeProps = {
      id: 'upgrade-ppc-2x',
      name: '2x PPC',
      cost: 100,
      currency: '£',
      onBuyClick: () => {}, // tslint:disable-line no-empty
    };
    upgrade = shallow(<Upgrade {...props} />);
  });

  it('renders', () => {
    expect(upgrade).toMatchSnapshot();
  });

  it('shows paper per click boost upgrade name and cost', () => {
      expect(upgrade.text()).toContain('2x PPC');
    expect(upgrade.text()).toContain('£100');
  });
});
