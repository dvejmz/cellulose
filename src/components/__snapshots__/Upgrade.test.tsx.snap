// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Upgrade /> renders 1`] = `
<div
  className="upgrade"
  data-test-id="upgrade-ppc-2x"
>
  <button
    className="btn"
    onClick={[Function]}
  >
    2x PPC
     (
    Â£
    100
    )
  </button>
</div>
`;
